src/example_robot_wrapper.cpp

```diff
@@ -94,20 +102,28 @@ ExampleRobotWrapper::~ExampleRobotWrapper()
 
 bool ExampleRobotWrapper::clearProtectiveStop()
 {
-  std::string safety_status;
-  dashboard_client_->commandSafetyStatus(safety_status);
-  bool is_protective_stopped = safety_status.find("PROTECTIVE_STOP") != std::string::npos;
-  if (is_protective_stopped)
+  if (primary_client_->isRobotProtectiveStopped())
   {
     URCL_LOG_INFO("Robot is in protective stop, trying to release it");
-    dashboard_client_->commandClosePopup();
-    dashboard_client_->commandCloseSafetyPopup();
-    if (!dashboard_client_->commandUnlockProtectiveStop())
+    if (dashboard_client_ != nullptr)
+    {
+      dashboard_client_->commandClosePopup();
+      dashboard_client_->commandCloseSafetyPopup();
+    }
+    try
+    {
+      primary_client_->commandUnlockProtectiveStop();
+    }
+    catch (const TimeoutException&)
     {
       std::this_thread::sleep_for(std::chrono::seconds(5));
-      if (!dashboard_client_->commandUnlockProtectiveStop())
+      try
       {
-        URCL_LOG_ERROR("Could not unlock protective stop");
+        primary_client_->commandUnlockProtectiveStop();
+      }
+      catch (const TimeoutException&)
+      {
+        URCL_LOG_ERROR("Robot did not unlock the protective stop within the given timeout");
```

Comment:
The user doesn't specify a timeout, so I prefer a slightly modification of the old log message. However, if you prefer the new log message I am also totally fine with keeping the new log message. 
```suggestion
        URCL_LOG_ERROR("Robot could not unlock the protective stop");
```